### PR TITLE
New structure for the 'Bonus Section'

### PR DESCRIPTION
**Pull request by VajraOfIndra**
The goals and rationales behind the choices are explained in Issue #788.

This pull-request focuses solely on navigation and content of the index pages ('Bonus Section', 'Bonus Raspberry Pi', 'Bonus Bitcoin' and 'Bonus Lightning' index pages). The content of the Bonus guides themselves has not been updated or modified.

The repo structure for the bonus section is:

|-- (main guide .md files, .yml, license, readme etc)
|-- bonus-section.md
|-- bonus
| |-- bitcoin
| | |-- bonus-bitcoin.md (index page)
| | |-- (bitcoin bonus guides .md files)
| |-- lightning
| | |-- bonus-lightning.md (index page)
| | |-- (lightning bonus guides .md files)
| |-- raspberry_pi
| | |-- raspberry-pi.md (index page)
| | |-- (raspberry pi bonus guides .md files)

For now, the guides are categorized under the following headings (headings to be added or changed as new bonus guides get merged):

* Bonus Raspberry Pi
  * Dashboards & Appearance
  * Security
  * Hardware
* Bonus Bitcoin
  * Electrum servers
  * Privacy
    * Coinjoin
  * Security
    * Multisig
* Bonus Lightning
  * Maintenance
  * Resilience
  * Dashboards & Wallets
    * CLI-only
    * GUI - Desktop
    * GUI - Mobile
* Even more extras

A web version of the PR can be viewed [HERE](https://vajraofindra.github.io/RaspiBolt/bonus/lightning/bonus-lightning.html)

### CODE REVIEW

**Comment by Stadicus** [comment]
Thanks, I'll properly review this later today :+1:

File: bonus/lightning/remote-lncli.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
+---
+layout: default
+title: LNTOP terminal dashboard
+parent: Bonus Lightning
+grand_parent: Bonus Section
+nav_exclude: true
+has_toc: false
+---
+## Bonus guide: lntop: LNTOP terminal dashboard
```
Comment (Stadicus) [inline bonus/lightning/remote-lncli.md:9]: The file `lightning/remote-lncli.md` somehow contains the "LNTOP" guide

File: bonus/lightning/auto-unlock.md [hunk 2]
```diff
@@ -1,12 +1,16 @@
 ---
 layout: default
 title: Auto-unlock LND
-parent: Bonus Section
-nav_order: 20
+parent: Bonus Lightning
+grand-parent: Bonus Section
```
Comment (Stadicus) [inline bonus/lightning/auto-unlock.md:5]: should be `grand_parent`, the breadcrumbs don't work like this

File: bonus/bitcoin/bonus-bitcoin.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+parent: Bonus Section
+has_children: false
+has_toc: false
+---
+
+## Electrum Servers
+* **[Electrum Personal Server](electrum-personal-server.md)** - an alternative to Electrs
+
+## Pivacy
+* *Coinjoin*
```
Comment (Stadicus) [inline bonus/bitcoin/bonus-bitcoin.md:14]: I think the "Coinjoin" title could be removed for now, too much structure for just one entry

File: bonus/bitcoin/bonus-bitcoin.md [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+  * **[JoinMarket on RaspiBolt](https://github.com/kristapsk/raspibolt-extras/blob/master/joinmarket.md)** - a CoinJoin software, which allows you to increase privacy and fungibility of on-chain Bitcoin transactions and also get a little passive income from your bitcoins
+
+## Security
+* *Multisig*
```
Comment (Stadicus) [inline bonus/bitcoin/bonus-bitcoin.md:18]: I think the "Multisig" title could be removed for now, too much structure for just one entry

File: bonus/lightning/static-backup-dropox.md [hunk 5]
```diff
@@ -1,12 +1,25 @@
 ---
```
Comment (Stadicus) [inline bonus/lightning/static-backup-dropox.md:1]: The filename `bonus/lightning/static-backup-dropox.md` contains a typo

**Review by Stadicus — CHANGES_REQUESTED** [review]
The overall structure looks great and can scale with many more guides.

Some thoughts on the Bonus subsection pages:
* The naming of the subsections (e.g., "Bonus Raspberry Pi" etc.) feels very redundant in the navigation. I know it's necessary to prevent the same categories as in the core guide. How about changing this to something like `+ Raspberry Pi`, `+ Bitcoin`, etc.?
* The Bonus subpages should contain a title, e.g., "Bonus Section: Raspberry Pi guides"
* Some horizontal rulers between the H2 sections could help visually divide them better
* For filenames, I used `-` to replace spaces. All directory and file names should follow this, e.g. for `./bonus/raspberry-pi/`
 

In general:
* For external links, `{:target="_blank"}` should be added to open it in a new browser tab
* also see some comments directly in the code

You can check out my updated and tested version of your branch here:  
https://stadicus.github.io/raspibolt-dev/bonus/bitcoin/bonus-bitcoin.html

You can see the individual changes in this commit. Technically it would be possible to import that into your own branch (e.g. with `merge` or `cherry-pick`), but it might be easier to do it manually.

https://github.com/Stadicus/raspibolt-dev/commit/e35d09e21d31622d9ff3065586a0ec525b0cdfc7

File: bonus/lightning/remote-lncli.md (hunk 1 above)
Comment (VajraOfIndra) [reply bonus/lightning/remote-lncli.md]: Oops, that must have been a copy/paste mistake, fixed!

File: bonus/lightning/auto-unlock.md (hunk 2 above)
Comment (VajraOfIndra) [reply bonus/lightning/auto-unlock.md]: fixed!

File: bonus/bitcoin/bonus-bitcoin.md (hunk 3 above)
Comment (VajraOfIndra) [reply bonus/bitcoin/bonus-bitcoin.md]: It makes sense. I opted for:

## Privacy (coinjoin)

and

## Security (multisig)

as I thought the Security categorisation of the Specter wallet guide was not obvious without the 'multisg' highlight....

But we could just use ' Privacy' and 'Security' if you prefer shorter/simpler heading, just let me know.

File: bonus/bitcoin/bonus-bitcoin.md (hunk 4 above)
Comment (VajraOfIndra) [reply bonus/bitcoin/bonus-bitcoin.md]: See above

File: bonus/lightning/static-backup-dropox.md (hunk 5 above)
Comment (VajraOfIndra) [reply bonus/lightning/static-backup-dropox.md]: thanks, fixed!

**Comment by VajraOfIndra** [comment]
> The overall structure looks great and can scale with many more guides.

Great :)

>     * The naming of the subsections (e.g., "Bonus Raspberry Pi" etc.) feels very redundant in the navigation. I know it's necessary to prevent the same categories as in the core guide. How about changing this to something like `+ Raspberry Pi`, `+ Bitcoin`, etc.?

Looks like a good solution indeed, I changed the file names and all paths using this new naming convention. I hope I didn't forget any.

>     * The Bonus subpages should contain a title, e.g., "Bonus Section: Raspberry Pi guides"

Agree, that's nicer! It's been added now.


>     * Some horizontal rulers between the H2 sections could help visually divide them better

Great suggestion, fixed.
 
>     * For filenames, I used `-` to replace spaces. All directory and file names should follow this, e.g. for `./bonus/raspberry-pi/`

Fixed.

>     * For external links, `{:target="_blank"}` should be added to open it in a new browser tab

Makes sense, fixed!
 
>     * also see some comments directly in the code

See below for a reply to each comment.

> You can check out my updated and tested version of your branch here: https://stadicus.github.io/raspibolt-dev/bonus/bitcoin/bonus-bitcoin.html
> 
> You can see the individual changes in this commit. Technically it would be possible to import that into your own branch (e.g. with `merge` or `cherry-pick`), but it might be easier to do it manually.

I didn't have time to setup git and learn how to use it so I opted to replicate all your changes manually. I went through the comments above and in the code and also reviwed your branch with the modifications. I think I got them all but I might have missed a couple, just let me know if you spot any typos or forgotten update.

File: bonus/bitcoin/bonus-bitcoin.md (hunk 3 above)
Comment (Stadicus) [reply bonus/bitcoin/bonus-bitcoin.md]: Sounds good, can always update later

**Review by Stadicus — CHANGES_REQUESTED** [review]
Great work, all my comments have been addressed.

One thing that can be improved: the main page of each part of the guide (as defined by the directory structure) should be called `index.md`, within the corresponding directory:

* /index.md
* /bonus/index.md
* /bonus/raspberry-pi/index.md

This ensures that links like the following, just specifying the directory, don't lead to a 404 error:

* https://vajraofindra.github.io/RaspiBolt/bonus/
* https://vajraofindra.github.io/RaspiBolt/bonus/bitcoin/

Everything else looks good! :tada:

**Comment by VajraOfIndra** [comment]
> One thing that can be improved: the main page of each part of the guide (as defined by the directory structure) should be called `index.md`, within the corresponding directory:
[...]
> This ensures that links like the following, just specifying the directory, don't lead to a 404 error:

Awesome, I updated accordingly. I noticed eralier that this was the recommended structure and naming convention in the 'Just the Docs' documentation but I didn't know about the advantage you mentionned above about 404 errors, that's neat.
 
> Everything else looks good! 🎉

Woohoo 🎉 :)

**Review by Stadicus — APPROVED** [review]
LGTM, tACK :)

@VajraOfIndra do you want to do the honors and push the "MERGE" button?

**Comment by VajraOfIndra** [comment]
> LGTM, tACK :)
> 
> @VajraOfIndra do you want to do the honors and push the "MERGE" button?

With great pleasure! :)

**Pull request merged**